dev-python/cadquery-ocp/cadquery-ocp-7.5.2_rc20210706.ebuild

```diff
@@ -0,0 +1,160 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{8..9} )
+
+# The "cmake" eclass is intentionally omitted here despite cmake being run.
+# That eclass expects a "${S}/CMakeLists.txt" file during src_prepare(),
+# whereas this package uniquely generates one "${BUILD_DIR}/OCP/CMakeLists.txt"
+# file for each active Python version.
+inherit llvm multiprocessing python-r1
+
+MY_COMMIT="9a6210e3c490e741719ccdada900c70e58b09c91"
+OCCT_PV=$(ver_cut 1-3)
+
+DESCRIPTION="Python wrapper for OCCT generated using pywrap"
+HOMEPAGE="https://github.com/CadQuery/OCP"
+SRC_URI="https://github.com/CadQuery/OCP/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+KEYWORDS="~amd64 ~x86"
+SLOT="0"
+
+# This version requirement derives from "${BUILD_DIR}/OCP/CMakeLists.txt".
+BDEPEND=">=dev-util/cmake-3.16"
+RDEPEND="~sci-libs/opencascade-${OCCT_PV}[json,tbb,vtk]"
```

Comment:
We need to take care here, vtk is being installed with python support enabled as well. The occt ebuild only requires vtk build with `USE=rendering` support enabled. AFAICS the build system requires the `WrappingPythonCore` vtk component being available.
This **will** restrict us to using python-single eclass, as vtk only supports a single python target. I think, it's non-trivial to add multi-python support to vtk, if it's possible at all.

Reply:
**Yup, yup, and more yup.** I just independently discovered this yesterday after adding an explicit dependency on VTK >= 9.0.0. Of course, everything then broke.

OCP now requires the same single Python target that VTK requires, which dramatically simplifies my life, which means this is the right thing to do.

> I think, it's non-trivial to add multi-python support to vtk, if it's possible at all.

Yeah... I kinda briefly contemplated that, then realized I needed to package [LIEF (Library for Instrumenting Executable Formats)](https://github.com/lief-project/LIEF) for this, then realized I was too tired to do anything more. A tired man is tired.